cmd/consul-dataplane/config.go

```diff
@@ -76,6 +77,24 @@ type ServiceFlags struct {
 	Partition     *string `json:"partition,omitempty"`
 }
 
+func (sf ServiceFlags) IsEmpty() bool {
+	return sf.NodeName == nil &&
+		sf.NodeID == nil &&
+		sf.ServiceID == nil &&
+		sf.ServiceIDPath == nil &&
+		sf.Namespace == nil &&
+		sf.Partition == nil
+}
+
+type ProxyFlags struct {
+	NodeName  *string `json:"nodeName,omitempty"`
+	NodeID    *string `json:"nodeID,omitempty"`
```

Comment:
Are these needed with v2. All workload ids must be unique within a tenant so it seems like we may not need these any longer.

Reply:
These are not needed with V2, however, these flags can be used for both v1 and v2. My intention is to deprecate old flags and to have one set of generically named flags that could be used against either API.